### PR TITLE
REST API small fixes

### PR DESCRIPTION
**Pull request by michalwski**
This PR addresses #1041, #1040, #1022

Proposed changes include:
* use `username` when creating a user via REST API
* update swagger spec

### CODE REVIEW

**Comment by Dzol** [comment]
I tried the `example` field in various places in the Swagger file but it doesn't seem to work. Search for _example_ at http://swagger.io/specification/ to find it. Can't link to it because there isn't one.

I thought it could help to say something like:

```JSON
{
  "name": "rabbit-hole",
  "owner": "alice@wonderland",
  "nick": "Ali"
}
```
 
Instead of just `"string"` under the example value.

**Comment by michalwski** [comment]
The thing is that "string" is taken from the field definition.

**Comment by michalwski** [comment]
I've managed to provide some examples in our swagger spec. Here's the result:
http://mongooseim.readthedocs.io/en/rest-api-small-fixes/swagger/index.html

File: test/ejabberd_tests/tests/muc_http_api_SUITE.erl [hunk 1]
```diff
@@ -133,8 +133,8 @@ send_message_to_room(Config) ->
         Host = <<"localhost">>,
         Path = <<"/mucs",$/,Host/binary,$/,Name/binary,$/,"messages">>,
```
Comment (gadgetci) [inline test/ejabberd_tests/tests/muc_http_api_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 134

Comment (gadgetci) [inline test/ejabberd_tests/tests/muc_http_api_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 134

Comment (gadgetci) [inline test/ejabberd_tests/tests/muc_http_api_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 134

Comment (gadgetci) [inline test/ejabberd_tests/tests/muc_http_api_SUITE.erl]: According to **Elvis**:
> Missing space right "," on line 134

File: doc/http-api/backend_swagger.yml [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
 info:
-  version: "0.0.0"
+  version: "1.0.0"
   title: MongooseIM control backend REST API
-  description: This REST API provides similar control functionaly to the `mongooseimctl` command. The response objects are modeled on the `ejabberd_commands` entries and print-outs from `mongooseimctl` though the final API will need to be consistent with the MongooseIM's **command registry**.
+  description: |
+    This REST API provides similar control functionaly to the `mongooseimctl` command.
+    The response objects are modeled on the `ejabberd_commands` entries and print-outs from `mongooseimctl` though the final API will need to be consistent with the MongooseIM's **command registry**.
+
+    In many cases fields like **user** or **caller** expect **JID** (jabber identity, f.e **alice@wonderland.com**).
```
Comment (Dzol) [inline doc/http-api/backend_swagger.yml]: Small detail but shall we use **username** instead of **user** so that it's consistent with the other changes?

Comment (michalwski) [inline doc/http-api/backend_swagger.yml]: Yes, it should be **username** I changed it in models and examples but forgot to change the description...

**Pull request merged**